package/gluon-core/files/lib/gluon/upgrade/200-wireless

```diff
@@ -10,11 +10,21 @@ if not sysconfig.gluon_version then
   uci:delete_all('wireless', 'wifi-iface')
 end
 
+local function get_channel(radio, config)
+  if uci:get_first('gluon-core', 'wireless', 'preserve_channels') then
+    return uci:get('wireless', radio, 'channel')
```

Comment:
I think the function should still fall back to config.channel if uci:get('wireless', radio, 'channel') returns nil, so previously unconfigured radios (maybe because of missing drivers... not supported at the moment, but we'd like to make the config more flexible) will get the correct channel even if preserve_channels is set. You can just add `or config.channel` to the return line above.